src/libs/actions/Report.js

```diff
@@ -551,15 +551,19 @@ function updateReportActionMessage(reportID, sequenceNumber, message) {
             return;
         }
 
-        // If the message is deleted, update the last read in case the deleted message is being counted in the unreadActionCount
+        // If the message is deleted, we should
+        // 1. update the last read in case the deleted message is being counted in the unreadActionCount
+        // 2. Get the previous lastMessageText and update it in Onyx
         setLocalLastRead(reportID, lastReadSequenceNumbers[reportID]);
-    });
-
-    // If this is the most recent message, update the lastMessageText in the report object as well
-    if (sequenceNumber === reportMaxSequenceNumbers[reportID]) {
         Onyx.merge(`${ONYXKEYS.COLLECTION.REPORT}${reportID}`, {
-            lastMessageText: ReportUtils.formatReportLastMessageText(message.text) || `[${Localize.translateLocal('common.deletedCommentMessage')}]`,
+            lastMessageText: ReportActions.getLastMessageText(reportID),
```

Comment:
I'm having trouble thinking of a better naming suggestion, but feels maybe not very obvious that `ReportActions .getLastMessageText()` would filter out deleted messages.

Reply:
Yea, I struggled with that too. Maybe `getLastVisibleMessageText` or `getLastRemainingMessageText`?

Reply:
I like `getLastVisibleMessageText()` there might be others we hide for reasons besides "deleting" in the future.